[MIG] stock_partner_delivery_window: Migration to 14.0

dzungtran89 opened the pull request:
Module dependency:

- [x] `base_time_window` https://github.com/OCA/server-tools/pull/1988
- [x] `partner_tz`: migrated in https://github.com/OCA/partner-contact/pull/1043

grindtildeath committed:
Add stock_partner_delivery_window

dzungtran89 committed:
[IMP] stock_partner_delivery_window: black, isort, prettier

dzungtran89 committed:
[MIG] stock_partner_delivery_window: Migration to 14.0

lmignon commented on test-requirements.txt:
```diff
@@ -0,0 +1,2 @@
+odoo14-addon-partner-tz @ git+https://github.com/trobz/partner-contact@14.0-mig-partner_tz#subdirectory=setup/partner_tz
```
@trobz can you add into your PR description than this one depends on others PR?

dzungtran89 replied:
@lmignon ok, I updated the PR description

simahawk replied:
it depends on this as well https://github.com/OCA/server-tools/pull/1988 (I've added it to description)

sebalix committed:
s_p_delivery_window: time format improvements

Two small improvements:

* add a hook to change the delivery time format
* no need to display seconds, hh:mm is enough

simahawk committed:
stock_partner_delivery_window: ease warning msg check/override

mmequignon committed:
stock_partner_delivery_window: Cleanup

mmequignon committed:
s_p_delivery_window: Add working days as delivery schedule preference

Stephanisk commented:
Hi,

Does someone have any news on the migration of this module to V14. I am really lost without this module.

Thx

simahawk commented on oca_dependencies.txt:
```diff
@@ -1 +1,3 @@
 # See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#oca_dependencies-txt
```
you can trash this file: [MQT=PIP](https://github.com/OCA/stock-logistics-workflow/blob/14.0/.travis.yml#L30)

dzungtran89 replied:
hi @simahawk , thx for your review. These changes have been cherry-picked to this MR

simahawk reviewed (changes requested):
LG overall. Would you mind cherry-picking these changes?

- [ ] https://github.com/OCA/stock-logistics-workflow/pull/822
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/824
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/830

simahawk commented:
> Hi,
> 
> Does someone have any news on the migration of this module to V14. I am really lost without this module.
> 
> Thx

You can help w/ functional and / or technical review to speed it up (same for the PRs that are dependencies for this one).
Meanwhile, you can use it from here. I suggest you give a try to https://pypi.org/project/git-aggregator/ to collect pending PRs alltogether.

Stephanisk commented:
I would love to help, just don't know how I could.

You said, meanwhile you can use it from here. What do you mean by this? I went to the aggregator website but I don't understand at all what I could do there/with that.

Sorry, I am learning more everyday, but this whole procedure on GitHub still gets me rattled when it comes to Odoo, it's complex.

simahawk commented:
You can also install modules directly from PRs. Add this to your requirements.txt:

```
odoo14-addon-partner_tz @ git+https://github.com/OCA/partner-contact@refs/pull/1043/head#subdirectory=setup/partner_tz
odoo14-addon-base_time_window @ git+https://github.com/OCA/server-tools@pull/1988/head#subdirectory=setup/base_time_window
odoo14-addon-stock_partner_delivery_window @ git+https://github.com/OCA/stock-logistics-workflow@refs/pull/775/head#subdirectory=setup/stock_partner_delivery_window
```

and run `pip install -r requirements.txt`. Then you can install modules as usual w/ `-i`.

Stephanisk commented:
Hi,

I am trying to wrap my head around it.


So I can install from updated pull requests no matter if the module is made
for v14?

For that I make a file requirements.txt with the content you placed and I
run a pip install.

I get lost now you say I can install modules as usual?? Where would I find
the module partner delivery window? Inside Odoo apps?

This is a production environment, if it would cause problems how will I
uninstall these modules?

Thanks for all your help so far. Send me your address please so I can send
you a thank you package .

Regards,
Stephane

On Sat, 19 Jun 2021, 09:39 Simone Orsi, ***@***.***> wrote:

> You can also install modules directly from PRs. Add this to your
> requirements.txt:
>
> odoo14-addon-partner_tz @ ***@***.***/pull/1043/head#subdirectory=setup/partner_tz
> odoo14-addon-base_time_window ***@***.***/pull/1043/head#subdirectory=setup/partner_tzodoo14-addon-base_time_window> @ ***@***.***/1988/head#subdirectory=setup/base_time_window
> odoo14-addon-stock_partner_delivery_window ***@***.***/1988/head#subdirectory=setup/base_time_windowodoo14-addon-stock_partner_delivery_window> @ ***@***.***/pull/775/head#subdirectory=setup/stock_partner_delivery_window
>
> and run pip install -r requirements.txt. Then you can install modules as
> usual w/ -i.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/stock-logistics-workflow/pull/775#issuecomment-864370576>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHWACDJ5ANQHZWWRKKAJH73TTRCTBANCNFSM4V7D4UYA>
> .
>

simahawk commented:
Once you run pip install as mentioned above you'll have the modules available for installing. 

I gave you this option because it avoids you the need of understanding and using git-aggregator.

If you are not familiar w/ python installations nor w/ git-aggregator nor w/ how to checkout PRs locally I'd recommend you ask for help to a professional (either for training of for technical work).

> This is a production environment, if it would cause problems how will I
uninstall these modules?

Never do this on prod envs if you don't know what you are doing and if you don't know how to rollback things.

You should try this locally 1st or on a test instance.

dzungtran89 committed:
[REF] Remove unused file

grindtildeath committed:
[FW][14.0] stock_partner_delivery_window: Fix default value on delivery_time_preference

The default value introduced by workdays feature breaks the installation
of Demo databases as the change is not reflected in sale_partner_delivery_window.

simahawk commented on stock_partner_delivery_window/models/res_partner.py:
```diff
@@ -0,0 +1,154 @@
+            ("workdays", "Weekdays (Monday to Friday)"),
+        ],
+        string="Delivery time schedule preference",
+        default="workdays",
```
@grindtildeath @mmequignon the default here should be changed?

grindtildeath replied:
Yes, please cherry-pick https://github.com/OCA/stock-logistics-workflow/pull/862 :pray:

hailangvn replied:
Thanks @simahawk , @grindtildeath for your comments. We are working on this.

hailangvn replied:
@simahawk, @grindtildeath, The PR has been updated.

simahawk reviewed (commented):
@grindtildeath @mmequignon please review

simahawk commented on test-requirements.txt:
```diff
@@ -0,0 +1,2 @@
+odoo14-addon-partner-tz @ git+https://github.com/trobz/partner-contact@14.0-mig-partner_tz#subdirectory=setup/partner_tz
```
you don't need these anymore

simahawk reviewed (approved):
LGTM but test requirements

Stephanisk commented:
Hi, I see Travis made test that didn't conflict. does this mean that the migration will soon be done.

Do you need any help in translations? I would be glad to help out

simahawk commented:
@grindtildeath and/or @mmequignon can you review?

simahawk commented:
/ocabot merge patch
